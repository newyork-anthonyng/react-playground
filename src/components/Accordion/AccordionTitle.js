import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';

const AccordionTitle = ({
  children,
  isOpen,
  ariaControls,
  onClick,
  id,
}) => (
  <dt role="heading" aria-level="3">
    <button
      aria-expanded={`${isOpen}`}
      aria-controls={ariaControls}
      onClick={onClick}
      id={id}
    >
      {children}
    </button>
  </dt>
);

AccordionTitle.defaultProps = {
  isOpen: false,
  onClick: noop,
};

AccordionTitle.propTypes = {
  children: T.node,
  isOpen: T.bool,
  ariaControls: T.string,
  id: T.string,
  onClick: T.func,
};

export default AccordionTitle;
