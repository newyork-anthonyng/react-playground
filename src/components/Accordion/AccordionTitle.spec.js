import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import noop from '../../utility/noop';
import AccordionTitle from './AccordionTitle';

const defaultProps = {
  isOpen: false,
  ariaControls: 'some-element-id',
  id: 'my-element-id',
  onClick: noop,
  onKeyDown: noop,
  buttonRef: noop,
};

it('should render correctly', () => {
  const wrapper = shallow(
    <AccordionTitle {...defaultProps}>
      <h1>Accordion Title</h1>
    </AccordionTitle>
  );

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it('should run callback when clicked', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <AccordionTitle
      {...defaultProps}
      onClick={cb}
    >
      <h1>Accordion Title</h1>
    </AccordionTitle>
  );

  const button = wrapper.find('button');
  button.simulate('click');

  expect(cb).toHaveBeenCalledTimes(1);
});

it('should run callback when pressing a key', () => {
  const cb = jest.fn();
  const wrapper = shallow(
    <AccordionTitle
      {...defaultProps}
      onKeyDown={cb}
    >
      <h1>Accordion Title</h1>
    </AccordionTitle>
  );

  const button = wrapper.find('button');
  button.simulate('keydown', { keyCode: 38 });

  expect(cb).toHaveBeenCalledTimes(1);
});
