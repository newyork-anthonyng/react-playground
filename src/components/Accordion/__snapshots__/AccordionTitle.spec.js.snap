// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<dt
  aria-level="3"
  role="heading"
>
  <button
    aria-controls="some-element-id"
    aria-expanded="false"
    id="my-element-id"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <h1>
      Accordion Title
    </h1>
  </button>
</dt>
`;
